### ISSUE TITLE
FR: Ability for a frame to indicate on initial load that storage access is required

### ISSUE THREAD

**Issue by lghall** [privacycg/storage-access#131]
At Google Workspace, we have embeds that require authentication on initial load (for example: an embedded document or a calendar appointment booking page).

This request is to expose a way for a frame to automatically request storage access on initial load via a browser-level UI associated with the frame (such as an overlay over the frame), and automatically reload the frame (and ideally other same-site frames on the same page in this state) if access is granted.

**Comment by annevk** [comment]
I don't think this is something WebKit would be interested in. We want end users to engage with the nested document and guide the user around the initial request.

**Comment by bvandersloot-mozilla** [comment]
Ditto from Mozilla- if the iframe doesn't have permission to get storage access, what should be shown? I think having an interstitial page (same-origin as the intended embedded document) that requests storage-access and redirects to the intended document is a better solution. This request won't even need user interaction if the permission is already set.

**Comment by johannhof** [comment]
So I think there may be some interesting things we could experiment with to help this request in spirit in the future, such as a dedicated permission element and / or more inline UI for Storage Access, but as it stands it's not high on our list of priorities either. I think I'd be interested in websites' real world experiences regarding the prompt-in-the-corner user experience and engagement rate to get a better sense of how much improvement here is needed.

I think this is still worth keeping around with a "future" label though, we've had a few issues stick around for a while to get eventually resolved (or improved).
